scripts/metrics/common/query_metrics.py

```diff
@@ -0,0 +1,157 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Provides query functionality for .csv metrics from S3 buckets
+"""
+
+import argparse
+import csv
+import os
+
+from boto3.session import Session
+
+BUCKET_NAME = "o3de-metrics"
+
+SESSION = None
+DOWNLOADED_FILES = []
+
+
+def _get_session():
+    """
+    Gets an AWS session via boto3 or returns the existing one
+    """
+    global SESSION
```

Comment:
I recommend directly initializing and passing the session context around, instead of managing a global singleton. Strange things could happen when this gets imported from multiple scripts.